File: C++/225. Implement Stack using Queues.cpp
class MyStack {
public:
    vector<int>a;
    MyStack() {
        
    }
    
    void push(int x) {
        a.push_back(x);
    }
    
    int pop() {
        int x=a.back();
        a.pop_back();
        return x;
    }
    
    int top() {
        return a.back();
    }
    
    bool empty() {
        return a.size()==0;
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: JAVA/225. Implement Stack using Queues.java
class MyStack {
    ArrayList<Integer>a;
    public MyStack() {
        a=new ArrayList<>();
    }
    
    public void push(int x) {
        a.add(x);
    }
    
    public int pop() {
        int x=a.get(a.size()-1);
        a.remove(a.size()-1);
        return x;
    }
    
    public int top() {
        return a.get(a.size()-1);
    }
    
    public boolean empty() {
        return a.size()==0;
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */